extensions/OpenGraphMeta/OpenGraphMeta.php

```diff
@@ -1,132 +1,132 @@
-function efSetMainImagePH( $out, $parserOutput, $data ) {
-	$out->mMainImage = wfFindFile( Title::newFromDBkey($data['dbkey'], NS_FILE) );
-}
-
-$wgHooks['BeforePageDisplay'][] = 'efOpenGraphMetaPageHook';
-function efOpenGraphMetaPageHook( OutputPage &$out, &$sk ) {
-	global $wgLogo, $wgSitename, $wgXhtmlNamespaces, $egFacebookAppId, $egFacebookAdmins;
-	$wgXhtmlNamespaces["og"] = "http://opengraphprotocol.org/schema/";
-	$title = $out->getTitle();
-	$isMainpage = $title->isMainPage();
-
-	$meta = array();
-
-	if ( $isMainpage ) {
-		$meta["og:type"] = "website";
-		$meta["og:title"] = $wgSitename;
-	} else {
-		$meta["og:type"] = "article";
-		$meta["og:site_name"] = $wgSitename;
-		// Try to chose the most appropriate title for showing in news feeds.
-		if ( ( defined('NS_BLOG_ARTICLE') && $title->getNamespace() == NS_BLOG_ARTICLE ) ||
-			( defined('NS_BLOG_ARTICLE_TALK') && $title->getNamespace() == NS_BLOG_ARTICLE_TALK ) ) {
-			$meta["og:title"] = $title->getSubpageText();
-		} elseif (defined('NS_USER_WALL_MESSAGE') && $title->getNamespace() == NS_USER_WALL_MESSAGE) {
-			$meta["og:title"] = $out->mPagetitle;
-		} else {
-			$meta["og:title"] = $title->getText();
-		}
-	}
-
-	if ( isset( $out->mMainImage ) && ( $out->mMainImage !== false ) ) {
-		if( is_object( $out->mMainImage ) ){
-			$meta["og:image"] = wfExpandUrl($out->mMainImage->createThumb(100*3, 100));
-		} else {
-			// In some edge-cases we won't have defined an object but rather a full URL.
-			$meta["og:image"] = $out->mMainImage;
-		}
-	} elseif ( $isMainpage ) {
-		$meta["og:image"] = wfExpandUrl($wgLogo);
-	}
-	if ( isset($out->mDescription) ) { // set by Description2 extension, install it if you want proper og:description support
-		$meta["og:description"] = $out->mDescription;
-	}
-	$meta["og:url"] = $title->getFullURL();
-	if ( $egFacebookAppId ) {
-		/* begin wikia change */
-		// $meta["fb:app_id"] = $egFacebookAppId;
-		// fb:app_id needs a prefix property declaring the namespace, so just add it directly
-		$out->addHeadItem("meta:property:fb:app_id", "	".Html::element( 'meta', array( 'property' => 'fb:app_id', 'content' => $egFacebookAppId, 'prefix' => "fb: http://www.facebook.com/2008/fbml" ) )."\n");
-		/* end wikia change */
-	}
-	if ( $egFacebookAdmins ) {
-		$meta["fb:admins"] = $egFacebookAdmins;
-	}
-	/* begin wikia change */
-	Hooks::run( 'OpenGraphMetaHeaders', array( "meta"=>&$meta, "title"=>$title ) );
-	/* end wikia change */
-	foreach( $meta as $property => $value ) {
-		if ( $value ) {
-			if ( isset( OutputPage::$metaAttrPrefixes ) && isset( OutputPage::$metaAttrPrefixes['property'] ) ) {
-				$out->addMeta( "property:$property", $value );
-			} else {
-				$out->addHeadItem("meta:property:$property", "	".Html::element( 'meta', array( 'property' => $property, 'content' => $value ) )."\n");
-			}
-		}
-	}
-
-	return true;
-}
-
-$egFacebookAppId = null;
-$egFacebookAdmins = null;
-
+<?php
+/**
+ * OpenGraphMeta
```

Comment:
It might be worth checking if we can kill this extension and use our Open Graph service.